kafka/consume.go

```diff
@@ -90,14 +95,45 @@ func (r *ConfluentConsumer) Seek(offset int64, timeout time.Duration) error {
 		return errors.New("Timeout should be set when calling Seek")
 	}
 
-	err := r.c.Seek(kafkalib.TopicPartition{Topic: &r.conf.Topic, Offset: kafkalib.Offset(offset)}, int(timeoutMs))
+	var err error
+	if r.partition < 0 {
+		err = r.c.Seek(kafkalib.TopicPartition{Topic: &r.conf.Topic, Offset: kafkalib.Offset(offset)}, int(timeoutMs))
+	} else {
+		err = r.c.Seek(kafkalib.TopicPartition{Topic: &r.conf.Topic, Partition: r.partition, Offset: kafkalib.Offset(offset)}, int(timeoutMs))
+	}
 	if err, ok := err.(kafkalib.Error); ok && err.Code() == kafkalib.ErrTimedOut {
 		return ErrSeekTimedOut
 	}
 
 	return err
 }
 
+// SeekToTime seeks to the specified time
+func (r *ConfluentConsumer) SeekToTime(t time.Time, timeout time.Duration) error {
+	timeoutMs := timeout.Milliseconds()
+	if timeoutMs == 0 {
+		// Otherwise the call will be asynchronous, losing error handling.
+		return errors.New("Timeout should be set when calling Seek")
+	}
+
+	var offsets []kafkalib.TopicPartition
+	var err error
+	millisSinceEpoch := t.UnixNano() / 1000000
+	if r.partition < 0 {
+		offsets, err = r.c.OffsetsForTimes([]kafka.TopicPartition{{Topic: &r.conf.Topic, Offset: kafkalib.Offset(millisSinceEpoch)}}, int(timeoutMs))
+	} else {
+		offsets, err = r.c.OffsetsForTimes([]kafka.TopicPartition{{Topic: &r.conf.Topic, Partition: r.partition, Offset: kafkalib.Offset(millisSinceEpoch)}}, int(timeoutMs))
```

Comment:
what happens if the time is out of range of what kafka holds?
i remember there was an option to specify if you want to earliest or the latest offset in that case. something like that?

Reply:
The seek will return and error